packages/ui-components/src/components/select-multi-options/select-multi-options.helper.ts

```diff
@@ -1,16 +1,56 @@
-import { isEmpty } from 'lodash';
-import { ISelectMultiOptions, ISelectMultiOption } from './select-multi-options.types';
+import { ISelectMultiOption, ISelectMultiOptions } from './select-multi-options.types';
+import { EToggleState, ISelectOption } from '../../types';
 
-export const getFlattenedOptionsMap = (options: ISelectMultiOptions): Record<string, ISelectMultiOption> => {
-	if (!options || Object.values(options).length === 0) {
-		return {};
+const buildSelectOption = (optionKey: string, options: ISelectMultiOptions = {}, selectedOptions: Record<string, boolean> = {}): ISelectOption => {
+	const option = options[optionKey];
+	const childrenOptions = buildSelectOptions(option.options, selectedOptions);
+
+	return {
+		...option,
+		options: childrenOptions,
+		selected: selectedOptions[optionKey] === true,
+		togglable: true,
+		state: getOptionToggleState(option, selectedOptions)
+	};
+};
+
+const getOptionToggleState = (option: ISelectMultiOption, selectedOptions: Record<string, boolean> = {}): EToggleState => {
+	const children = Object.values(option.options ?? {});
+	if (children.length === 0) {
+		if (selectedOptions[option.value]) {
+			return EToggleState.Selected;
+		}
+
+		return EToggleState.None;
+	}
+
+	const childrenStates = children.map(childrenOpt => getOptionToggleState(childrenOpt, selectedOptions));
+
+	if (childrenStates.every(childrenState => childrenState === EToggleState.Selected)) {
+		return EToggleState.Selected;
+	}
+
+	if (childrenStates.every(childrenState => childrenState === EToggleState.Indeterminate)) {
```

Comment:
isn't enough 1 item to be `Indeterminated` to this return true? I said this because in the previous condition, we check all the items are selected, so this one seems more redundant.

Reply:
Yes, I'll do a little refactor here, thank you.